### DISCUSSION TITLE
BIRT export to .xlsx

### DISCUSSION THREAD

**Discussion by sumitpuri** [eclipse-birt/birt#1148] [Q&A] [answered]
Does BIRT support export to .xlsx? If yes, which version does that?

<img width="490" alt="image" src="https://user-images.githubusercontent.com/6925028/208751103-ab970b37-04bb-4446-bf54-a59eeb4457f4.png">

**Comment by speckyspooky** [comment, marked as answer]
BIRT suport Excel-XLSX since BIRT version 4.6 based on the Spudsoft-implementation.
Please use one of the latest versions (e.g. BIRT 4.12) of BIRT and you will get the option of XLSX.

**Reply by 858882847** [reply]
![226176527-24b85c97-f1b0-425b-a2cb-42d8429e4b44](https://user-images.githubusercontent.com/117696273/226236146-af41df36-343d-462b-b775-f3c0a6af5491.png)
![226176528-239538e4-8c4f-46d7-9b6c-07abc02ad5ab](https://user-images.githubusercontent.com/117696273/226236152-31fe87c3-6171-44e1-acab-8a18882949ca.png)
Is it a backend configuration issue for BIRT to export the height and width ratio columns in xlsx and xls files

**Reply by 858882847** [reply]
[wind.zip](https://github.com/eclipse/birt/files/11013304/wind.zip)

**Reply by speckyspooky** [reply]
Please remove the table with of 100% which is currently given on your table.
Because the width is calculated based on the print layout of the table. The column width will be work without the table width.

The result of Spudsoft XLS & Spudsoft XLSX looks like the same.

Next hint, please check your script because the style value none must be a string like: "none",
therefore you get the error message at preview side.

![2023-03-20 12_52_35-workspace - C__Users_gutma_Downloads_wind_Sbysd2 rptdesign - BIRT Report Designe](https://user-images.githubusercontent.com/41593722/226332298-e9249972-f7f6-4fc2-91b2-ff1343b7dc10.png)

**Reply by 858882847** [reply]
Thank you very much. My problem has been resolved. I don't know which direction Birt will take in the future. I am currently using Birt as an Excel transmitter

**Reply by 858882847** [reply]
> Please remove the table with of 100% which is currently given on your table. Because the width is calculated based on the print layout of the table. The column width will be work without the table width.
> 
> The result of Spudsoft XLS & Spudsoft XLSX looks like the same.
> 
> Next hint, please check your script because the style value none must be a string like: "none", therefore you get the error message at preview side.
> 
> ![2023-03-20 12_52_35-工作区 - C__Users_gutma_Downloads_wind_Sbysd2 rptdesign - BIRT 报告设计](https://user-images.githubusercontent.com/41593722/226332298-e9249972-f7f6-4fc2-91b2-ff1343b7dc10.png)

Due to the customer's requirement that there be three tables in a BIRT and that the BIRT be in triplicate, how to solve the problem of only one column of xlsx report output
[zcbh.zip](https://github.com/eclipse/birt/files/11026442/zcbh.zip)

**Reply by 858882847** [reply]
BIRT exports xlsx data in one 'Sheet' without multiple 'Sheets'

**Reply by 858882847** [reply]
[test.zip](https://github.com/eclipse/birt/files/11037167/test.zip)

**Reply by 858882847** [reply]
Originally, I added an extra layer of table outside these three layers of tables to control the pagination of the tables inside. This way, I could include the footer data of each table on each page of the pdf. However, now that I have exported the xlsx file, the three tables have been reduced to a single row without any hierarchical structure. So, I removed the outer layer of the table and the three tables now have a hierarchical structure. But when I export it as a pdf file, the footer of each table is only output on the last page.

**Reply by hvbtup** [reply]
With a background of creating BIRT reports since 2008,  here's my advice:

Usually you should use two different rptdesign files if you want to create a PDF representation and Excel representation of the same data.

These output formats are way too different, and the emitters have their own strenghs and weaknesses, too.
So you should not try to handle them in the same report.

The two reports can share a lot of functionality, of course. That's what *.rptlibrary and *.js files are for.
For example, you can move the data sets to a common library.

**Reply by hvbtup** [reply]
@858882847 You are always mixing different topcis over several issues. This is annoying. Personally, I'll stop answering your questions for a while.

**Reply by 858882847** [reply]
"Sorry, I will pay attention next time and create a new one again."
